docs/enhancements/add-external-providers.md

```diff
@@ -0,0 +1,182 @@
+---
+title: Add-Providers-Support
+authors:
+  - "@Gal-Zaidman"
+creation-date: 2021-07-25
+last-updated: 2021-07-25
+---
+
+# Add Providers Support
+
+## Summary
+
+Currently, the assisted installer supports Bare-Metal installations only.
+Openshift can be installed on various providers such as:
+- Red Hat Virtualization.
+- OpenStack.
+- VSphere.
+- AWS.
+- GCP.
+- Azure.
+
+This enhancement will define a clear interface to extend the assisted installer and allow
+provider installations.
+
+## Motivation
+
+Assisted Installer can be a great tool to perform UPI (User Provisioned) installation on external providers,
+providing the user with clear UI to see its provisioned machines joining the cluster, and lowering installation
+resources requirements.
+
+Assisted Installer is also ideal for mixed env configurations.
+** The RHV provider supports installing a mixed env where the Control-Plain (Master) nodes are deployed on top of RHV
+and Compute (Worker) nodes are Bare-Metal nodes.
+
+### Goals
+
+- Create a Provider interface, and hooks in the assisted installer code which separates the assisted installer core code
+ from the provider-specific code.
+- Add clear documentation for each provider specific function.
+- Document steps to install a provider cluster with assisted installer
+
+### Non-Goals
+
+- Support IPI installations.
+- Integrating all relevant [Openshift installer types](https://github.com/openshift/installer/tree/master/pkg/types)
+into the assisted installer code - this will require a large refactor of the assisted installer code.
+- Remove Baremetal from not appropriate stuct/functions names(example in open questions).
+ 
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As a RHV/VSphere admin I want to create an Openshift cluster on my RHV engine by using the assisted installer.
+
+#### Story 2
+
+As a RHV admin I want to create an Openshift cluster on my RHV engine by using the assisted installer in which
+Control-Plain (Master) nodes are deployed on top of RHV and Compute (Worker) nodes are Bare-Metal nodes.
+
+#### Story 3
+
+As an Openshift developer working on platform X I want to add support for installing Openshift with assisted installer
+on top of platform X.
+
+### Implementation Details/Notes/Constraints
+
+#### Changes to the assisted-service project:
+
+##### Step 1: Detect platform specific logic
+
+Go over the assisted installer flow and detect parts of the code which require provider specific logic.
```

Comment:
Today we rely on the "Manufacturer" that is returned by our inventory

Reply:
I didn't mean to detect the platform (this is step 4).
Here I meant to identify parts of the code that should be under the provider interface

Reply:
I see, a research step.  OK.